File: VB/DialogsDemo.Wpf/ViewModels/ThemedWindowModule/HeaderModels/CustomHeaderItemModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO

Namespace DialogsDemo
    Public Class CustomHeaderItemModel
        Inherits BaseHeaderItemModel

        Protected Overridable ReadOnly Property MessageBoxService() As IMessageBoxService
            Get
                Return Nothing
            End Get
        End Property
        Public Property Content() As String

        Public Sub Click()
            MessageBoxService.Show(String.Format("{0} command executed!", Content), "Demo", MessageBoxButton.OK, MessageBoxImage.Information, MessageBoxResult.OK)
        End Sub
        Public Sub New(ByVal index As Integer)
            ResourceKey = "CustomHeaderItem"
            Content = "Item " & index
        End Sub
    End Class
End Namespace

File: CS/DialogsDemo.Wpf/ViewModels/ThemedWindowModule/HeaderModels/CustomHeaderItemModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;

namespace DialogsDemo {
    public class CustomHeaderItemModel : BaseHeaderItemModel {
        protected virtual IMessageBoxService MessageBoxService { get { return null; } }
        public string Content { get; set; }

        public void Click() { MessageBoxService.Show(String.Format("{0} command executed!", Content), "Demo", MessageBoxButton.OK, MessageBoxImage.Information, MessageBoxResult.OK); }
        public CustomHeaderItemModel(int index) {
            ResourceKey = "CustomHeaderItem";
            Content = "Item " + index;
        }
    }
}
